### PR TITLE
Don't replace tabs with spaces in strings

### PR DESCRIPTION
**Pull request by mohe2015**
Maybe I didn't check correctly but to me it seems this still does replacement outside of strings through other means.

`git grep -P "\t" | grep ".nix:"` showed 92 remaining tabs in formatted nixpkgs which all look to be in strings. Unformatted where 134 tabs.

It seemed though that this had other formatting side-effects (Probably because tab characters are now counted differently). These definitely need to be investigated before merging.

Fixes https://github.com/nix-community/nixpkgs-fmt/issues/251

Need to add test of the fixed issue if this fix is actually acceptable.

### CODE REVIEW

**Comment by mohe2015** [comment]
```bash
git remote add 0x4A6F https://github.com/0x4A6F/nixpkgs
git fetch 0x4a6f master-fmt
git checkout upstream/master
git checkout -b nixpkgs-fmt-pr-270
git cherry-pick 113dce22c3cc048fdebab1cb1a6c5de20a6be9e6
nix shell github:mohe2015/nixpkgs-fmt/dont-replace-tabs-in-strings#nixpkgs-fmt github:NixOS/nixpkgs#treefmt
treefmt
git add .
git commit
exit

git checkout upstream/master
git checkout -b nixpkgs-fmt-master
git cherry-pick 113dce22c3cc048fdebab1cb1a6c5de20a6be9e6
nix shell github:nix-community/nixpkgs-fmt#nixpkgs-fmt github:NixOS/nixpkgs#treefmt
treefmt
git add .
git commit
```

git diff nixpkgs-fmt-master > a.diff
```diff
diff --git a/nixos/modules/security/sudo.nix b/nixos/modules/security/sudo.nix
index a1903260d19..6f0b9e1237d 100644
--- a/nixos/modules/security/sudo.nix
+++ b/nixos/modules/security/sudo.nix
@@ -220,8 +220,8 @@ in
           lists.flatten (
             map (
               rule: if (length rule.commands != 0) then [
-                (map (user: "${toUserString user}  ${rule.host}=(${rule.runAs})  ${toCommandsString rule.commands}") rule.users)
-                (map (group: "${toGroupString group}  ${rule.host}=(${rule.runAs})  ${toCommandsString rule.commands}") rule.groups)
+                (map (user: "${toUserString user}	${rule.host}=(${rule.runAs})	${toCommandsString rule.commands}") rule.users)
+                (map (group: "${toGroupString group}	${rule.host}=(${rule.runAs})	${toCommandsString rule.commands}") rule.groups)
               ] else []
             ) cfg.extraRules
           )
diff --git a/nixos/modules/services/backup/rsnapshot.nix b/nixos/modules/services/backup/rsnapshot.nix
index f7fbdf029eb..19416c14cce 100644
--- a/nixos/modules/services/backup/rsnapshot.nix
+++ b/nixos/modules/services/backup/rsnapshot.nix
@@ -5,16 +5,16 @@ with lib;
 let
   cfg = config.services.rsnapshot;
   cfgfile = pkgs.writeText "rsnapshot.conf" ''
-    config_version  1.2
-    cmd_cp  ${pkgs.coreutils}/bin/cp
-    cmd_rm  ${pkgs.coreutils}/bin/rm
-    cmd_rsync  ${pkgs.rsync}/bin/rsync
-    cmd_ssh  ${pkgs.openssh}/bin/ssh
-    cmd_logger  ${pkgs.inetutils}/bin/logger
-    cmd_du  ${pkgs.coreutils}/bin/du
-    cmd_rsnapshot_diff  ${pkgs.rsnapshot}/bin/rsnapshot-diff
-    lockfile  /run/rsnapshot.pid
-    link_dest  1
+    config_version	1.2
+    cmd_cp	${pkgs.coreutils}/bin/cp
+    cmd_rm	${pkgs.coreutils}/bin/rm
+    cmd_rsync	${pkgs.rsync}/bin/rsync
+    cmd_ssh	${pkgs.openssh}/bin/ssh
+    cmd_logger	${pkgs.inetutils}/bin/logger
+    cmd_du	${pkgs.coreutils}/bin/du
+    cmd_rsnapshot_diff	${pkgs.rsnapshot}/bin/rsnapshot-diff
+    lockfile	/run/rsnapshot.pid
+    link_dest	1
 
     ${cfg.extraConfig}
   '';
@@ -32,9 +32,9 @@ in
       extraConfig = mkOption {
         default = "";
         example = ''
-          retains  hourly  24
-          retain  daily  365
-          backup  /home/  localhost/
+          retains	hourly	24
+          retain	daily	365
+          backup	/home/	localhost/
         '';
         type = types.lines;
         description = ''
diff --git a/nixos/modules/services/networking/pptpd.nix b/nixos/modules/services/networking/pptpd.nix
index e4cc567ea78..50405c0d6cc 100644
--- a/nixos/modules/services/networking/pptpd.nix
+++ b/nixos/modules/services/networking/pptpd.nix
@@ -106,8 +106,8 @@ with lib;
           [ -f "$secrets" ] || cat > "$secrets" << EOF
           # From: pptpd-1.4.0/samples/chap-secrets
           # Secrets for authentication using CHAP
-          # client  server  secret    IP addresses
-          #username  pptpd  password  *
+          # client	server	secret		IP addresses
+          #username	pptpd	password	*
           EOF
 
           chown root.root "$secrets"
diff --git a/nixos/modules/services/networking/xl2tpd.nix b/nixos/modules/services/networking/xl2tpd.nix
index f3542c56d09..03db12401b1 100644
--- a/nixos/modules/services/networking/xl2tpd.nix
+++ b/nixos/modules/services/networking/xl2tpd.nix
@@ -114,8 +114,8 @@ with lib;
 
           [ -f ppp/chap-secrets ] || cat > ppp/chap-secrets << EOF
           # Secrets for authentication using CHAP
-          # client  server  secret    IP addresses
-          #username  xl2tpd  password  *
+          # client	server	secret		IP addresses
+          #username	xl2tpd	password	*
           EOF
 
           chown root.root ppp/chap-secrets
diff --git a/pkgs/applications/editors/vim/macvim.nix b/pkgs/applications/editors/vim/macvim.nix
index 696ad2def73..2c13a242d85 100644
--- a/pkgs/applications/editors/vim/macvim.nix
+++ b/pkgs/applications/editors/vim/macvim.nix
@@ -141,7 +141,7 @@ stdenv.mkDerivation {
   # Xcode project or pass it as a flag to xcodebuild as well.
   postConfigure = ''
     substituteInPlace src/auto/config.mk \
-      --replace "PERL_CFLAGS  =" "PERL_CFLAGS  = -I${darwin.libutil}/include" \
+      --replace "PERL_CFLAGS	=" "PERL_CFLAGS	= -I${darwin.libutil}/include" \
       --replace " -L${stdenv.cc.libc}/lib" "" \
       --replace " -L${darwin.libobjc}/lib" "" \
       --replace " -L${darwin.libunwind}/lib" "" \
diff --git a/pkgs/applications/networking/cluster/kubespy/default.nix b/pkgs/applications/networking/cluster/kubespy/default.nix
index 808c04ae22a..fa0e62c53fe 100644
--- a/pkgs/applications/networking/cluster/kubespy/default.nix
+++ b/pkgs/applications/networking/cluster/kubespy/default.nix
@@ -17,10 +17,10 @@ buildGoModule rec {
 
   # TODO: enable after https://github.com/pulumi/kubespy/issues/72 is addressed.
   # postInstall = ''
-  #   for shell in bash zsh; do
-  #     $out/bin/kubespy completion $shell > kubespy.$shell
-  #     installShellCompletion kubespy.$shell
-  #   done
+  # 	for shell in bash zsh; do
+  # 		$out/bin/kubespy completion $shell > kubespy.$shell
+  # 		installShellCompletion kubespy.$shell
+  # 	done
   # '';
 
   meta = with lib; {
diff --git a/pkgs/applications/version-management/arch/default.nix b/pkgs/applications/version-management/arch/default.nix
index dabdff10da7..c161d8446ac 100644
--- a/pkgs/applications/version-management/arch/default.nix
+++ b/pkgs/applications/version-management/arch/default.nix
@@ -18,8 +18,8 @@ stdenv.mkDerivation rec {
   # Instead of GNU Autoconf, tla uses Tom Lord's now
   # defunct `package-framework'.
   buildPhase = ''
-    mkdir +build && cd +build &&    \
-    ../src/configure --prefix="$out" &&    \
+    mkdir +build && cd +build &&		\
+    ../src/configure --prefix="$out" &&		\
     make install
   '';
 
diff --git a/pkgs/applications/window-managers/windowmaker/dockapps/wmsm-app.nix b/pkgs/applications/window-managers/windowmaker/dockapps/wmsm-app.nix
index 2c87c5b1fdd..1513d7b9433 100644
--- a/pkgs/applications/window-managers/windowmaker/dockapps/wmsm-app.nix
+++ b/pkgs/applications/window-managers/windowmaker/dockapps/wmsm-app.nix
@@ -23,7 +23,7 @@ stdenv.mkDerivation rec {
 
   postPatch = ''
     substituteInPlace Makefile \
-      --replace "PREFIX  = /usr/X11R6/bin" "" \
+      --replace "PREFIX	= /usr/X11R6/bin" "" \
       --replace "/usr/bin/install" "install"
   '';
 
diff --git a/pkgs/build-support/dotnetenv/build-solution.nix b/pkgs/build-support/dotnetenv/build-solution.nix
index 09c21108348..d81829b361b 100644
--- a/pkgs/build-support/dotnetenv/build-solution.nix
+++ b/pkgs/build-support/dotnetenv/build-solution.nix
@@ -31,55 +31,55 @@ stdenv.mkDerivation {
   '';
 
   installPhase = ''
-      addDeps()
-      {
-    if [ -f $1/nix-support/dotnet-assemblies ]
-    then
-        for i in $(cat $1/nix-support/dotnet-assemblies)
+        addDeps()
+        {
+    	if [ -f $1/nix-support/dotnet-assemblies ]
+    	then
+    	    for i in $(cat $1/nix-support/dotnet-assemblies)
+    	    do
+    		windowsPath=$(cygpath --windows $i)
+    		assemblySearchPaths="$assemblySearchPaths;$windowsPath"
+
+    		addDeps $i
+    	    done
+    	fi
+        }
+
+        for i in ${toString assemblyInputs}
         do
-      windowsPath=$(cygpath --windows $i)
-      assemblySearchPaths="$assemblySearchPaths;$windowsPath"
+    	windowsPath=$(cygpath --windows $i)
+    	echo "Using assembly path: $windowsPath"
 
-      addDeps $i
-        done
-    fi
-      }
-
-      for i in ${toString assemblyInputs}
-      do
-    windowsPath=$(cygpath --windows $i)
-    echo "Using assembly path: $windowsPath"
-
-    if [ "$assemblySearchPaths" = "" ]
-    then
-        assemblySearchPaths="$windowsPath"
-    else
-        assemblySearchPaths="$assemblySearchPaths;$windowsPath"
-    fi
+    	if [ "$assemblySearchPaths" = "" ]
+    	then
+    	    assemblySearchPaths="$windowsPath"
+    	else
+    	    assemblySearchPaths="$assemblySearchPaths;$windowsPath"
+    	fi
 
-    addDeps $i
-      done
+    	addDeps $i
+        done
 
-      echo "Assembly search paths are: $assemblySearchPaths"
+        echo "Assembly search paths are: $assemblySearchPaths"
 
-      if [ "$assemblySearchPaths" != "" ]
-      then
-    echo "Using assembly search paths args: $assemblySearchPathsArg"
-    export AssemblySearchPaths=$assemblySearchPaths
-      fi
+        if [ "$assemblySearchPaths" != "" ]
+        then
+    	echo "Using assembly search paths args: $assemblySearchPathsArg"
+    	export AssemblySearchPaths=$assemblySearchPaths
+        fi
 
-      mkdir -p $out
-      MSBuild.exe ${toString slnFile} /nologo /t:${targets} /p:IntermediateOutputPath=$(cygpath --windows $out)\\ /p:OutputPath=$(cygpath --windows $out)\\ /verbosity:${verbosity} ${options}
+        mkdir -p $out
+        MSBuild.exe ${toString slnFile} /nologo /t:${targets} /p:IntermediateOutputPath=$(cygpath --windows $out)\\ /p:OutputPath=$(cygpath --windows $out)\\ /verbosity:${verbosity} ${options}
 
-      # Because .NET assemblies store strings as UTF-16 internally, we cannot detect
-      # hashes. Therefore a text files containing the proper paths is created
-      # We can also use this file the propagate transitive dependencies.
+        # Because .NET assemblies store strings as UTF-16 internally, we cannot detect
+        # hashes. Therefore a text files containing the proper paths is created
+        # We can also use this file the propagate transitive dependencies.
 
-      mkdir -p $out/nix-support
+        mkdir -p $out/nix-support
 
-      for i in ${toString assemblyInputs}
-      do
-          echo $i >> $out/nix-support/dotnet-assemblies
-      done
+        for i in ${toString assemblyInputs}
+        do
+            echo $i >> $out/nix-support/dotnet-assemblies
+        done
   '';
 }
diff --git a/pkgs/build-support/dotnetenv/wrapper.nix b/pkgs/build-support/dotnetenv/wrapper.nix
index 96459a16a85..cd9e2fd3f7c 100644
--- a/pkgs/build-support/dotnetenv/wrapper.nix
+++ b/pkgs/build-support/dotnetenv/wrapper.nix
@@ -28,37 +28,37 @@ dotnetenv.buildSolution {
   slnFile = "Wrapper.sln";
   assemblyInputs = [ application ];
   preBuild = ''
-      addRuntimeDeps()
-      {
-    if [ -f $1/nix-support/dotnet-assemblies ]
-    then
-        for i in $(cat $1/nix-support/dotnet-assemblies)
-        do
-      windowsPath=$(cygpath --windows $i | sed 's|\\|\\\\|g')
-      assemblySearchArray="$assemblySearchArray @\"$windowsPath\""
+        addRuntimeDeps()
+        {
+    	if [ -f $1/nix-support/dotnet-assemblies ]
+    	then
+    	    for i in $(cat $1/nix-support/dotnet-assemblies)
+    	    do
+    		windowsPath=$(cygpath --windows $i | sed 's|\\|\\\\|g')
+    		assemblySearchArray="$assemblySearchArray @\"$windowsPath\""
 
-      addRuntimeDeps $i
-        done
-    fi
-      }
+    		addRuntimeDeps $i
+    	    done
+    	fi
+        }
 
-      export exePath=$(cygpath --windows $(find ${application} -name \*.exe) | sed 's|\\|\\\\|g')
+        export exePath=$(cygpath --windows $(find ${application} -name \*.exe) | sed 's|\\|\\\\|g')
 
-      # Generate assemblySearchPaths string array contents
-      for path in ${toString assemblyInputs}
-      do
-          assemblySearchArray="$assemblySearchArray @\"$(cygpath --windows $path | sed 's|\\|\\\\|g')\", "
-    addRuntimeDeps $path
-      done
+        # Generate assemblySearchPaths string array contents
+        for path in ${toString assemblyInputs}
+        do
+            assemblySearchArray="$assemblySearchArray @\"$(cygpath --windows $path | sed 's|\\|\\\\|g')\", "
+    	addRuntimeDeps $path
+        done
 
-      sed -e "s|@ROOTNAMESPACE@|${namespace}Wrapper|" \
-          -e "s|@ASSEMBLYNAME@|${namespace}|" \
-          Wrapper/Wrapper.csproj.in > Wrapper/Wrapper.csproj
+        sed -e "s|@ROOTNAMESPACE@|${namespace}Wrapper|" \
+            -e "s|@ASSEMBLYNAME@|${namespace}|" \
+            Wrapper/Wrapper.csproj.in > Wrapper/Wrapper.csproj
 
-      sed -e "s|@NAMESPACE@|${namespace}|g" \
-          -e "s|@MAINCLASSNAME@|${mainClassName}|g" \
-    -e "s|@EXEPATH@|$exePath|g" \
-    -e "s|@ASSEMBLYSEARCHPATH@|$assemblySearchArray|" \
-          Wrapper/Wrapper.cs.in > Wrapper/Wrapper.cs
+        sed -e "s|@NAMESPACE@|${namespace}|g" \
+            -e "s|@MAINCLASSNAME@|${mainClassName}|g" \
+    	-e "s|@EXEPATH@|$exePath|g" \
+    	-e "s|@ASSEMBLYSEARCHPATH@|$assemblySearchArray|" \
+            Wrapper/Wrapper.cs.in > Wrapper/Wrapper.cs
   '';
 }
diff --git a/pkgs/build-support/rust/build-rust-crate/log.nix b/pkgs/build-support/rust/build-rust-crate/log.nix
index 3a15ddfa76a..5f01ce1785f 100644
--- a/pkgs/build-support/rust/build-rust-crate/log.nix
+++ b/pkgs/build-support/rust/build-rust-crate/log.nix
@@ -50,12 +50,12 @@ in
   '';
 
   noisily = colors: verbose: ''
-    noisily() {
-      ${lib.optionalString verbose ''
-        echo_colored -n "Running "
-        echo $@
-      ''}
-      $@
-    }
+      noisily() {
+    	  ${lib.optionalString verbose ''
+          echo_colored -n "Running "
+          echo $@
+    	  ''}
+    	  $@
+      }
   '';
 }
diff --git a/pkgs/common-updater/generic-updater.nix b/pkgs/common-updater/generic-updater.nix
index 7db0b47d75b..8483f9bbd1d 100644
--- a/pkgs/common-updater/generic-updater.nix
+++ b/pkgs/common-updater/generic-updater.nix
@@ -59,7 +59,7 @@ let
 
     # print available tags
     for tag in $tags; do
-        echo "#  found $pname version: $tag" >> ${fileForGitCommands}
+        echo "#	found $pname version: $tag" >> ${fileForGitCommands}
     done
 
     # cut any revision prefix not used in the NixOS package version
diff --git a/pkgs/development/compilers/bigloo/default.nix b/pkgs/development/compilers/bigloo/default.nix
index c2fe5bec036..29e26aa9153 100644
--- a/pkgs/development/compilers/bigloo/default.nix
+++ b/pkgs/development/compilers/bigloo/default.nix
@@ -38,15 +38,15 @@ stdenv.mkDerivation rec {
 
   patchPhase = ''
     # Fix absolute paths.
-    sed -e 's=/bin/mv=mv=g' -e 's=/bin/rm=rm=g'      \
-        -e 's=/tmp=$TMPDIR=g' -i autoconf/*    \
-        [Mm]akefile*   */[Mm]akefile*   */*/[Mm]akefile*  \
-        */*/*/[Mm]akefile*   */*/*/*/[Mm]akefile*    \
+    sed -e 's=/bin/mv=mv=g' -e 's=/bin/rm=rm=g'			\
+        -e 's=/tmp=$TMPDIR=g' -i autoconf/*		\
+        [Mm]akefile*   */[Mm]akefile*   */*/[Mm]akefile*	\
+        */*/*/[Mm]akefile*   */*/*/*/[Mm]akefile*		\
         comptime/Cc/cc.scm gc/install-*
 
     # Make sure we don't change string lengths in the generated
     # C files.
-    sed -e 's=/bin/rm=     rm=g' -e 's=/bin/mv=     mv=g'  \
+    sed -e 's=/bin/rm=     rm=g' -e 's=/bin/mv=     mv=g'	\
         -i comptime/Cc/cc.c
   '';
 
diff --git a/pkgs/development/compilers/factor-lang/default.nix b/pkgs/development/compilers/factor-lang/default.nix
index c3be52ce4b4..63135fb4b80 100644
--- a/pkgs/development/compilers/factor-lang/default.nix
+++ b/pkgs/development/compilers/factor-lang/default.nix
@@ -89,7 +89,7 @@ stdenv.mkDerivation rec {
     (echo $(cat $TMPDIR/so.lst | wc -l) "libs found in cache \`/etc/ld.so.cache'";
     for l in $(<$TMPDIR/so.lst);
     do
-      echo "  $(basename $l) (libc6,x86-64) => $l";
+      echo "	$(basename $l) (libc6,x86-64) => $l";
     done)> $out/lib/factor/ld.so.cache
 
     sed -ie "s#/sbin/ldconfig -p#cat $out/lib/factor/ld.so.cache#g" \
diff --git a/pkgs/development/compilers/mlton/20130715.nix b/pkgs/development/compilers/mlton/20130715.nix
index 67a44738a57..2cc901050a0 100644
--- a/pkgs/development/compilers/mlton/20130715.nix
+++ b/pkgs/development/compilers/mlton/20130715.nix
@@ -65,7 +65,7 @@ stdenv.mkDerivation rec {
 
     substituteInPlace $(pwd)/../${usr_prefix}/bin/mlton --replace '/${usr_prefix}/lib/mlton' $(pwd)/../${usr_prefix}/lib/mlton
   '' + lib.optionalString stdenv.cc.isClang ''
-    sed -i "s_  patch -s -p0 <gdtoa.hide-public-fns.patch_  patch -s -p0 <gdtoa.hide-public-fns.patch\n\tsed -i 's|printf(emptyfmt|printf(\"\"|g' ./gdtoa/arithchk.c_" ./runtime/Makefile
+    sed -i "s_	patch -s -p0 <gdtoa.hide-public-fns.patch_	patch -s -p0 <gdtoa.hide-public-fns.patch\n\tsed -i 's|printf(emptyfmt|printf(\"\"|g' ./gdtoa/arithchk.c_" ./runtime/Makefile
   '' + lib.optionalString stdenv.isDarwin ''
     sed -i 's|XCFLAGS += -I/usr/local/include -I/sw/include -I/opt/local/include||' ./runtime/Makefile
   '';
diff --git a/pkgs/development/interpreters/python/cpython/default.nix b/pkgs/development/interpreters/python/cpython/default.nix
index 8186bc10f67..034bad69e49 100644
--- a/pkgs/development/interpreters/python/cpython/default.nix
+++ b/pkgs/development/interpreters/python/cpython/default.nix
@@ -364,7 +364,7 @@ with passthru; stdenv.mkDerivation {
   ] ++ optional static "LDFLAGS=-static";
 
   preConfigure = ''
-    for i in /usr /sw /opt /pkg; do  # improve purity
+    for i in /usr /sw /opt /pkg; do	# improve purity
       substituteInPlace ./setup.py --replace $i /no-such-path
     done
   '' + optionalString stdenv.isDarwin ''
diff --git a/pkgs/development/libraries/box2d/default.nix b/pkgs/development/libraries/box2d/default.nix
index d9d848c9d74..350f8f36cb8 100644
--- a/pkgs/development/libraries/box2d/default.nix
+++ b/pkgs/development/libraries/box2d/default.nix
@@ -34,7 +34,7 @@ stdenv.mkDerivation rec {
 
   prePatch = ''
     substituteInPlace Box2D/Common/b2Settings.h \
-      --replace 'b2_maxPolygonVertices  8' 'b2_maxPolygonVertices  15'
+      --replace 'b2_maxPolygonVertices	8' 'b2_maxPolygonVertices	15'
   '';
 
   meta = with lib; {
diff --git a/pkgs/development/libraries/opendmarc/default.nix b/pkgs/development/libraries/opendmarc/default.nix
index 4e81769b123..0dedfa3a85e 100644
--- a/pkgs/development/libraries/opendmarc/default.nix
+++ b/pkgs/development/libraries/opendmarc/default.nix
@@ -17,7 +17,7 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [ autoreconfHook makeWrapper ];
 
   postPatch = ''
-    substituteInPlace configure.ac --replace '  docs/Makefile' ""
+    substituteInPlace configure.ac --replace '	docs/Makefile' ""
     patchShebangs contrib reports
   '';
 
diff --git a/pkgs/development/lisp-modules/quicklisp-to-nix-overrides.nix b/pkgs/development/lisp-modules/quicklisp-to-nix-overrides.nix
index 5c45d66ab85..16d121f3c34 100644
--- a/pkgs/development/lisp-modules/quicklisp-to-nix-overrides.nix
+++ b/pkgs/development/lisp-modules/quicklisp-to-nix-overrides.nix
@@ -146,8 +146,8 @@ in
     overrides = y: (x.overrides y) // {
       postPatch = ''
         patch <<EOD
-        --- swank-loader.lisp  2017-08-30 16:46:16.554076684 -0700
-        +++ swank-loader-new.lisp  2017-08-30 16:49:23.333450928 -0700
+        --- swank-loader.lisp	2017-08-30 16:46:16.554076684 -0700
+        +++ swank-loader-new.lisp	2017-08-30 16:49:23.333450928 -0700
         @@ -155,7 +155,7 @@
                           ,(unique-dir-name)))
             (user-homedir-pathname)))
diff --git a/pkgs/development/ocaml-modules/mlgmpidl/default.nix b/pkgs/development/ocaml-modules/mlgmpidl/default.nix
index 8a819585b0d..a0026e3a8ca 100644
--- a/pkgs/development/ocaml-modules/mlgmpidl/default.nix
+++ b/pkgs/development/ocaml-modules/mlgmpidl/default.nix
@@ -20,7 +20,7 @@ stdenv.mkDerivation rec {
 
   postConfigure = ''
     sed -i Makefile \
-      -e 's|^  /bin/rm |  rm |'
+      -e 's|^	/bin/rm |	rm |'
     mkdir -p $out/lib/ocaml/${ocaml.version}/site-lib/stublibs
   '';
 
diff --git a/pkgs/development/tools/misc/cflow/default.nix b/pkgs/development/tools/misc/cflow/default.nix
index 1fdffc4f7f7..e724acdef2a 100644
--- a/pkgs/development/tools/misc/cflow/default.nix
+++ b/pkgs/development/tools/misc/cflow/default.nix
@@ -10,8 +10,8 @@ stdenv.mkDerivation rec {
   };
 
   patchPhase = ''
-    substituteInPlace "src/cflow.h"          \
-      --replace "/usr/bin/cpp"            \
+    substituteInPlace "src/cflow.h"					\
+      --replace "/usr/bin/cpp"						\
                 "$(cat ${stdenv.cc}/nix-support/orig-cc)/bin/cpp"
   '';
 
diff --git a/pkgs/development/tools/neoload/default.nix b/pkgs/development/tools/neoload/default.nix
index d210d0d2b71..51c9cd42509 100644
--- a/pkgs/development/tools/neoload/default.nix
+++ b/pkgs/development/tools/neoload/default.nix
@@ -18,8 +18,8 @@ let
   dotInstall4j = path: writeTextFile {
     name = "dot-install4j";
     text = ''
-      JRE_VERSION  ${path}  1  7  0  7
-      JRE_INFO  ${path}  94
+      JRE_VERSION	${path}	1	7	0	7
+      JRE_INFO	${path}	94
     '';
   };
 
diff --git a/pkgs/games/xconq/default.nix b/pkgs/games/xconq/default.nix
index f28b6db9bcc..73ffad373e7 100644
--- a/pkgs/games/xconq/default.nix
+++ b/pkgs/games/xconq/default.nix
@@ -48,11 +48,11 @@ stdenv.mkDerivation rec {
 
   patchPhase = ''
     # Fix Makefiles
-    find . -name 'Makefile.in' -exec sed -re 's@^        ( *)(cd|[&][&])@  \1\2@' -i '{}' ';'
+    find . -name 'Makefile.in' -exec sed -re 's@^        ( *)(cd|[&][&])@	\1\2@' -i '{}' ';'
     find . -name 'Makefile.in' -exec sed -e '/chown/d; /chgrp/d' -i '{}' ';'
     # do not set sticky bit in nix store
     find . -name 'Makefile.in' -exec sed -e 's/04755/755/g' -i '{}' ';'
-    sed -e '/^      * *[$][(]tcltkdir[)]\/[*][.][*]/d' -i tcltk/Makefile.in
+    sed -e '/^			* *[$][(]tcltkdir[)]\/[*][.][*]/d' -i tcltk/Makefile.in
 
     # Fix C files
     sed -re 's@[(]int[)]color@(long)color@' -i tcltk/tkmap.c
diff --git a/pkgs/os-specific/linux/bionic-prebuilt/default.nix b/pkgs/os-specific/linux/bionic-prebuilt/default.nix
index ecfae9cf6b5..8f11a2fdf45 100644
--- a/pkgs/os-specific/linux/bionic-prebuilt/default.nix
+++ b/pkgs/os-specific/linux/bionic-prebuilt/default.nix
@@ -81,8 +81,8 @@ stdenvNoCC.mkDerivation rec {
     sed -i 's,union semun {,union Xsemun {,' $out/include/linux/sem.h
     sed -i 's,struct __kernel_sockaddr_storage,#define sockaddr_storage __kernel_sockaddr_storage\nstruct __kernel_sockaddr_storage,' $out/include/linux/socket.h
     sed -i 's,#ifndef __UAPI_DEF_.*$,#if 1,' $out/include/linux/libc-compat.h
-    substituteInPlace $out/include/linux/in.h --replace "__be32    imr_" "struct in_addr    imr_"
-    substituteInPlace $out/include/linux/in.h --replace "__be32    imsf_" "struct in_addr    imsf_"
+    substituteInPlace $out/include/linux/in.h --replace "__be32		imr_" "struct in_addr		imr_"
+    substituteInPlace $out/include/linux/in.h --replace "__be32		imsf_" "struct in_addr		imsf_"
     substituteInPlace $out/include/linux/sysctl.h --replace "__unused" "_unused"
 
     # what could possibly live in <linux/compiler.h>
diff --git a/pkgs/tools/X11/xmacro/default.nix b/pkgs/tools/X11/xmacro/default.nix
index 158df7b0799..cf0ef6d8bf8 100644
--- a/pkgs/tools/X11/xmacro/default.nix
+++ b/pkgs/tools/X11/xmacro/default.nix
@@ -9,7 +9,7 @@ stdenv.mkDerivation rec {
     sha256 = "1p9jljxyn4j6piljiyi2xv6f8jhjbzhabprp8p0qmqxaxgdipi61";
   };
 
-  preInstall = "echo -e 'install:\n  mkdir \${out}/bin;\n  cp xmacrorec2 xmacroplay \${out}/bin;' >>Makefile; ";
+  preInstall = "echo -e 'install:\n	mkdir \${out}/bin;\n	cp xmacrorec2 xmacroplay \${out}/bin;' >>Makefile; ";
 
   buildInputs = [ libX11 libXtst xorgproto libXi ];
 
diff --git a/pkgs/tools/admin/tightvnc/default.nix b/pkgs/tools/admin/tightvnc/default.nix
index aae30130a37..f539bb430fe 100644
--- a/pkgs/tools/admin/tightvnc/default.nix
+++ b/pkgs/tools/admin/tightvnc/default.nix
@@ -65,8 +65,8 @@ stdenv.mkDerivation rec {
         -e 's@unix/:7100@'$fontPath'@' \
         -i vncserver
 
-    sed -e 's@.* CppCmd .*@#define CppCmd    cpp@' -i Xvnc/config/cf/linux.cf
-    sed -e 's@.* CppCmd .*@#define CppCmd    cpp@' -i Xvnc/config/cf/Imake.tmpl
+    sed -e 's@.* CppCmd .*@#define CppCmd		cpp@' -i Xvnc/config/cf/linux.cf
+    sed -e 's@.* CppCmd .*@#define CppCmd		cpp@' -i Xvnc/config/cf/Imake.tmpl
     sed -i \
         -e 's@"uname","xauth","Xvnc","vncpasswd"@"uname","Xvnc","vncpasswd"@g' \
         -e "s@\<xauth\>@${xauth}/bin/xauth@g" \
diff --git a/pkgs/tools/graphics/argyllcms/default.nix b/pkgs/tools/graphics/argyllcms/default.nix
index f0f4918c947..b36e84b10fe 100644
--- a/pkgs/tools/graphics/argyllcms/default.nix
+++ b/pkgs/tools/graphics/argyllcms/default.nix
@@ -79,7 +79,7 @@ stdenv.mkDerivation rec {
         USE_LIBUSB = false ;
 
         # Compile in graph plotting code (Not fully implemented)
-        USE_PLOT = true ;    # [true]
+        USE_PLOT = true ;		# [true]
 
         JPEGLIB = ;
         JPEGINC = ;
diff --git a/pkgs/tools/misc/man-db/default.nix b/pkgs/tools/misc/man-db/default.nix
index 03100b2635d..f1739cbd50d 100644
--- a/pkgs/tools/misc/man-db/default.nix
+++ b/pkgs/tools/misc/man-db/default.nix
@@ -24,10 +24,10 @@ stdenv.mkDerivation rec {
     sed -i 's/^MANDATORY_MANPATH/# &/' src/man_db.conf.in
 
     # Add Nix-related manpaths
-    echo "MANPATH_MAP  /nix/var/nix/profiles/default/bin  /nix/var/nix/profiles/default/share/man" >> src/man_db.conf.in
+    echo "MANPATH_MAP	/nix/var/nix/profiles/default/bin	/nix/var/nix/profiles/default/share/man" >> src/man_db.conf.in
 
     # Add mandb locations for the above
-    echo "MANDB_MAP  /nix/var/nix/profiles/default/share/man  /var/cache/man/nixpkgs" >> src/man_db.conf.in
+    echo "MANDB_MAP	/nix/var/nix/profiles/default/share/man	/var/cache/man/nixpkgs" >> src/man_db.conf.in
   '';
 
   configureFlags = [

```

**Comment by mohe2015** [comment]
See https://github.com/NixOS/nixpkgs/pull/146798#issuecomment-974710993 for the results of the rebuild. To me it seems this could work. What do you think?

**Comment by zimbatm** [comment]
It was added for a reason but I need to look back in the git history

**Comment by mohe2015** [comment]
I could image you could create an example with strategically placed tabs that doesn't format properly now / doesn't remove the tabs. Maybe I will look into it more tomorrow if you haven't found the reason.

**Comment by r-burns** [comment]
I agree with not formatting tabs to spaces inside string literals, but I think we should still reformat nix expression indentation.

**Comment by mohe2015** [comment]
> I agree with not formatting tabs to spaces inside string literals, but I think we should still reformat nix expression indentation.

If I ran my tests correctly in realistic code (and one existing testcase in this repo) this still works but I agree this probably needs some correction for edge cases or at least test cases. If I did the nixpkgs format diff correctly no issue occured in nixpkgs though.

**Comment by mohe2015** [comment]
The commit message needs to be fixed though.

**Comment by r-burns** [comment]
By that I mean, if I write a .nix file that uses tabs as indentation, nixpkgs-fmt should (at least by default) convert them to spaces. Does that still happen? I thought that was only done by this pre-pass.

**Comment by mohe2015** [comment]
> By that I mean, if I write a .nix file that uses tabs as indentation, nixpkgs-fmt should (at least by default) convert them to spaces. Does that still happen? I thought that was only done by this pre-pass.

https://github.com/nix-community/nixpkgs-fmt/blob/master/src/lib.rs#L176 still works so in some sense yes. I'm not familiar with the code though to estimate if there are any other realistic use cases in which it works / doesn't work.

**Comment by zimbatm** [comment]
We're hitting this issue again where rnix-parser doesn't give us the indent as a separate token for multiline strings (and comments). Otherwise it would be easier to only replace \t everywhere except the string content.

**Comment by mohe2015** [comment]
> We're hitting this issue again where rnix-parser doesn't give us the indent as a separate token for multiline strings (and comments). Otherwise it would be easier to only replace \t everywhere except the string content.

Okay now I think I'm getting closer to what the actual problem is. Though I'm not sure why we can't skip replacing tabs in strings and do it everywhere else because I thought nix wouldn't see tabs as indent in Multiline Strings anyways (will check later).

**Comment by mohe2015** [comment]
I'm not sure though if that is rnix-parsers responsibility to me it more feels like it's an uncommon use case

**Comment by mohe2015** [comment]
Brainstorming is happening in https://github.com/nix-community/nixpkgs-fmt/pull/275

**Comment by mohe2015** [comment]
Closing for https://github.com/nix-community/nixpkgs-fmt/pull/275
